### DISCUSSION TITLE
Unable to view ssh authentication success/failures in Wazuh dashboard Security Events.

### DISCUSSION THREAD

**Discussion by wazuhnoob** [wazuh/wazuh#13926] [Q&A] [answered]
Hi, 
I have tried adding rules to the localrules.xml from here https://github.com/wazuh/wazuh-ruleset/blob/master/rules/0095-sshd_rules.xml , 
but i am still unable to see any authentication failures or access when i try to ssh from my host. 
Can anyone please advise. Thank you.

**Comment by mikykeane** [comment, marked as answer]
Hello @wazuhnoob , 

you should not need to add any rules to be able to see authentication failures or success as all SSH rules and decoders are on Wazuh by default. 

First of all, are you able to see any alerts? Can you see any type of event? 

Is the host that you are monitoring connected to the manager? Showing up as connected and all?

My recommendation is to remove the rules you have added to `localrules.xml` as those will surely interfere with the existing ones. Likely causing an error which could be the reason why you are not seeing any alerts. Then restart the Wazuh Manager to ensure the changes are applied. 

Then, once you have everything operative. Lets first go to the `Wazuh Dashboard -> Security Events -> Events`
 
![image](https://user-images.githubusercontent.com/4438081/174840775-55bb3afb-9f2a-44b8-8fe8-fbf2d07306bd.png)

Here, you will be able to see all the events you have received. Make sure you have no additional filters added and that you don't have any agents pinned. Do you see any alerts at all?  

If you don't see any alerts, then most likely the issue is either on Filebeat or Elastic. 

Lets go to the Wazuh manager and check that the service is up and running: 

```
   systemctl status wazuh-manager
   systemctl status filebeat
```

If everything seems fine, let's check the alerts directly in Wazuh:

```

tail /var/ossec/logs/alerts/alerts,json 

```

If you are seeing alerts there that are not in the Dashboards (look at the timestamp), then we can determine that Wazuh is working fine and the issue is on either Filebeat or Elasticsearch. 


Now, if these suggestions and restarting the services did not help, please send me the following: 
`/var/ossec/logs/ossec.log`
`/var/log/elasticserach/<cluster-name>.log`
`systemctl status filebeat -l`

Let me know if any of this helped and send me back the requested information. I will allow us to have some insight on what could the issue be. 

Best regards, 
Miki Keane

**Reply by wazuhnoob** [reply]
Hi sir, 
thanks for the reply. Please see my input:
The agent is connected to manager, i can see alerts when i restart my agent and i tried a few audit rules and they worked.
I have removed the rules from local_rules.xml and the events are still not showing.
Both wazuh-manager and filebeat is working.
I have attached the files you requested, i cannot find the /var/log/elastic as i am running standalone.
[ossec.log](https://github.com/wazuh/wazuh/files/8953821/ossec.log)
[filebeat.txt](https://github.com/wazuh/wazuh/files/8953822/filebeat.txt)

**Reply by mikykeane** [reply]
Hello @wazuhnoob , 

apologies as I meant `Wazuh Indexer`. Which from versions 4.3 and above, is taking over Elasticsearch. 

The logs will be in `/var/log/wazuh-indexer/<name>.log`. Apologies for any confusion. 


First of all, looking at your logs I see an issue on `local_rules.xml`. It is running into a CRITICAL error: 
```
2022/06/22 01:38:44 wazuh-analysisd: ERROR: (1226): Error reading XML file 'etc/rules/local_rules.xml': XMLERR: Element not opened. (line 68).
2022/06/22 01:38:44 wazuh-analysisd: CRITICAL: (1220): Error loading the rules: 'etc/rules/local_rules.xml'.
2022/06/22 01:40:04 wazuh-analysisd: ERROR: (1226): Error reading XML file 'etc/rules/local_rules.xml': XMLERR: Element not opened. (line 68).
2022/06/22 01:40:04 wazuh-analysisd: CRITICAL: (1220): Error loading the rules: 'etc/rules/local_rules.xml'.

```

This is typically because Local Rules, usually has a sample content. And when it tries to load an empty file, it encounters the error above. 

To restore the default rules, you could add the following: 

```
<!-- Local rules -->

<!-- Modify it at your will. -->
<!-- Copyright (C) 2016, Wazuh Inc. -->

<!-- Example -->
<group name="local,syslog,sshd,">

  <!--
  Dec 10 01:02:02 host sshd[1234]: Failed none for root from 1.1.1.1 port 1066 ssh2
  -->
  <rule id="100001" level="5">
    <if_sid>5716</if_sid>
    <srcip>1.1.1.1</srcip>
    <description>sshd: authentication failed from IP 1.1.1.1.</description>
    <group>authentication_failed,pci_dss_10.2.4,pci_dss_10.2.5,</group>
  </rule>

</group>
```

Save the file and restart. 

This shold solve the issues when loading the ruleset. 

Now, you mentioned receiving some alerts. Did you see those alerts on the Wazuh Dashboards? Such as the Wazuh restart for example?

What were the contents of `/var/ossec/logs/alerts/alerts.json`? Everything on that file should be on the Wazuh Dashboard, so if that is the case, Filebeat and the Wazuh Indexer are working as expected. 

Now, gathering logins and logoffs (among many other Use Cases) should be captured by default upon installation. If you have changed the default configuration, then your agent or manager, might not be capturing the necessary logs to analyze. 

Make sure that this configuration is on both your Manager and your agent: 

```
<localfile>
  <location>/var/log/secure</location>
  <log_format>syslog</log_format>
</localfile>
```

Here is some documentation on how [Logcolletion ](https://documentation.wazuh.com/current/user-manual/capabilities/log-data-collection/how-it-works.html#how-it-works) works.

Let me know if you get your Authentication success and failure events comming in with this. And, to be sure, you have not changed anything on the default rules path, right? On `/var/ossec/ruleset/`. That folder should always be left untouched. 

Best regards, 
MIguel Keane

**Reply by wazuhnoob** [reply]
Hi Miguel,
I have the cluster log here for your reference.
[wazuh-cluster.log](https://github.com/wazuh/wazuh/files/8956413/wazuh-cluster.log)

I have replaced my local_rules.xml with the content you shared above.
For agent events, i am only able to see stuff like ossec agent stopped, ossec agent started but i am still unable to view the events for 
For the /var/ossec/logs/alerts/alerts.json file, yes i can see the events (ssh etc) on the Wazuh dashboard but only for the manager.
I have added the configuration you mentioned to ossec.conf on both manager and agent but still unable to see any ssh authentication logs.
I am also unable to see any events for my Windows agent (RDP authentication). 
Lastly, yes i have not touched anything on the default rules path.

Thank you again for your assistance.

**Reply by mikykeane** [reply]
Hi @wazuhnoob , 

Apologies for the late response, I was OOO for most of last week. 

Alright. Can you confirm that the agent is successfully connected?

Right now, if you can confirm that all alerts showing in `/var/ossec/logs/alerts/alerts.json` are also showing on the Wazuh Dashboard, the issue is not from events not being correctly parsed to the Indexer. 

This leaves us with 2 main possibilities: 

1. **(Most likely)** The Wazuh agent is not sending the events to the Wazuh manager: Connectivity issues, changes to the default configuration (by default, the Wazuh agent is configured to read from relevant logsources and Eventchannel to receive all security relevant information).    

2. The events are being received by the Wazuh Manager, but are not being correctly analyzed. Given this that you sent: 
>  can see the events (ssh etc) on the Wazuh dashboard but only for the manager

We can safely assume that this is not the cause. As the rules are being correctly triggered for events from the Wazuh Manager. 


First of all. Is the Wazuh agent connected? If we go to the `Agents` section, we should see our agent as `active`:

![image](https://user-images.githubusercontent.com/4438081/176141139-f738598c-83e1-4b43-96a7-39a6f0bb44af.png)


If the agent is Active. Lets go to `Security Events` and `Explore agent`: 

![image](https://user-images.githubusercontent.com/4438081/176141933-726c7c9c-9919-4e9d-94bc-6017341199c6.png)


Select the Agent in question: 

![image](https://user-images.githubusercontent.com/4438081/176142024-a2f6a769-60fa-4171-98cf-85d093b28dd1.png)


Expand the time if needed. Can you see any alerts at all from the agent?

![image](https://user-images.githubusercontent.com/4438081/176142214-a0b53160-bc83-4f5d-b702-2427f56a1845.png)


If we can confirm that both the Agent is Active and that we are not receiving alerts from it, we need to continue the troubleshooting from the agent. 

Go to the agent, which I assume is a Windows machine from what you have said. 

Open the application: 

![image](https://user-images.githubusercontent.com/4438081/176142885-f2a8f875-bb93-4406-a59c-27c39115c282.png)

Just in case, lets Restart first:

![image](https://user-images.githubusercontent.com/4438081/176143261-ca64ef77-60ac-4f6c-a8c0-02086fefe574.png)

And please send me the following: 

Logs and the Configuration: 

![image](https://user-images.githubusercontent.com/4438081/176143399-fa4b3e3a-0f9c-4a41-83b5-b9704b25aa74.png)

Also just in case, send me the contents of `agent.conf`. 

![image](https://user-images.githubusercontent.com/4438081/176143641-50ac8f98-9a57-4eaa-9c37-7a9a751957b8.png)



Also, just for reference. Restarting the agent should have generated an alert as follows: 

![image](https://user-images.githubusercontent.com/4438081/176144151-184a6c0b-dc2a-40d3-8298-26afa8f1b722.png)

Also, this is how an RDP login alert looks in Wazuh with default rules and configuration: 

![image](https://user-images.githubusercontent.com/4438081/176144588-e1a58371-1c4b-4faa-ae46-3179491bc669.png)


Looking forward for the information requested. Hopefully we can get your environment up and running correctly ASAP. 

Best regards, 
Miguel Keane

**Reply by wazuhnoob** [reply]
Hi Miguel,
Thanks so much for your assistance. My environment is finally ok.

**Reply by mikykeane** [reply]
Hi @wazuhnoob , 

glad to hear that! Did you figure out what was wrong?

Regards, 
Miguel Keane

**Comment by wazuhnoob** [comment]
I realise my Wazuh does not come with elasticsearch. Could that be the reason? Do i hve to install elasticsearch separately?

**Reply by mikykeane** [reply]
Hello @wazuhnoob ,

apologies as I meant Wazuh Indexer. Which from versions 4.3 and above, is taking over Elasticsearch. 

So if you have Wazuh Indexer, you do not need Elasticsearch or OpenDistro at all. 

Best regards, 
Miguel Keane

**Reply by wazuhnoob** [reply]
Hi, 
Can someone please help. Thank you.
